### ISSUE TITLE
False positive on decorators that return a `staticmethod`

### ISSUE THREAD

**Issue by junrushao1994** [PyCQA/pylint#7300]
### Bug description

In some usecases, a decorator is applied to a method of a python class, and it returns a staticmethod instead. Those usecases are supported by mypy with a `TYPE_CHECKING` guard: https://github.com/python/mypy/issues/6726#issuecomment-486907505.

```python
# pylint: disable=missing-docstring,invalid-name
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    returns_staticmethod = staticmethod
else:
    def returns_staticmethod(f): ...


class MyClass:
    @returns_staticmethod
    def my_staticmethod(a, b): # <======== pylint (E0213): Method should have "self" as first argument
        return a + b
```

However, testing with pylint, an error pops up indicating `self` is not the first argument of the method above. Additionally, even if we are not using the `TYPE_CHECKING` guard, pylint does not seem to  propagate `staticmethod` decorator:

```python
# pylint: disable=missing-docstring,invalid-name,too-few-public-methods
my_decorator = staticmethod

class MyClass:
    @my_decorator
    def my_decorated_func(a, b): # <======== pylint (E0213): Method should have "self" as first argument
        return a + b
```

Ideally, if pylint propagates `staticmethod` through `my_decorator`, the false positive would go away.

Thanks in advance!

### Configuration

Configuration is not relevant

### Command used

```shell
pylint main.py
```

### Pylint output

```shell
************* Module main
main.py:14:4: E0213: Method should have "self" as first argument (no-self-argument)
```

### Expected behavior

Expected that pylint propagates `staticmethod` via `my_decorator` so that the snippets below will work without warning:

```python
# pylint: disable=missing-docstring,invalid-name,too-few-public-methods
my_decorator = staticmethod

class MyClass:
    @my_decorator
    def my_decorated_func(a, b): # <======== pylint (E0213): Method should have "self" as first argument
        return a + b
```

### Pylint version

```shell
>>> pylint --version
pylint 2.12.2
astroid 2.9.0
Python 3.8.13 (default, Mar 28 2022, 11:38:47)
[GCC 7.5.0]
```

### OS / Environment

Ubuntu 20.04

### Additional dependencies

N/A

**Comment by mbyrnepr2** [comment]
Looks like it can be fixed by checking `node.decoratornames`:
```python
(venv310) Marks-MacBook-Air-2:programming markbyrne$ pylint example.py 
> /Users/markbyrne/programming/pylint/pylint/checkers/classes/class_checker.py(1920)_check_first_arg_for_type()
-> if node.type == "staticmethod":
(Pdb) node.decoratornames
<BoundFunctionWrapper at 0x102d79c60 for method at 0x102d5f0c0>
(Pdb) node.decoratornames()
{'builtins.staticmethod'}
(Pdb) 
```
